### PR TITLE
Make add-gceinstance and add-gcemanagedinstancegroup throws non-terminating error

### PR DESCRIPTION
**Pull request by quoctruong**
This is to fix https://github.com/GoogleCloudPlatform/google-cloud-powershell/issues/586.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/588)

### CODE REVIEW

**Comment by iantalarico** [comment]
I'm a little confused here (probably due to my lack of PS knowledge).  Why are we only catching conflicts here?  What is the expected result of 'SilentlyContinue'?  Does this mean any failure results in non termination?

**Comment by quoctruong** [comment]
@iantalarico There are 2 types of errors here: terminating and non-terminating error. 'SilentlyContinue' can only suppress non-terminating error. For some cmdlets, I think it makes sense in some common scenarios to write a non-terminating error (like the instance that we are trying to create already exists) instead of throwing the terminating error. This way, the user can suppress it.

File: Google.PowerShell/Compute/GceInstanceCmdlets.cs [hunk 1]
```diff
@@ -630,11 +631,21 @@ protected override void ProcessRecord()
...
             {
-                WriteObject(Service.Instances.Get(Project, Zone, instance.Name).Execute());
-            });
+                Operation operation = request.Execute();
+                AddZoneOperation(Project, Zone, operation, () =>
+                {
+                    WriteObject(Service.Instances.Get(Project, Zone, instance.Name).Execute());
+                });
+            }
+            catch (GoogleApiException apiEx) when (apiEx.HttpStatusCode == HttpStatusCode.Conflict)
```
Comment (iantalarico) [inline Google.PowerShell/Compute/GceInstanceCmdlets.cs]: Should this also be based on 'SilentlyContinue'?

Comment (quoctruong) [reply Google.PowerShell/Compute/GceInstanceCmdlets.cs]: @iantalarico What do you mean by that?

Comment (ILMTitan) [reply Google.PowerShell/Compute/GceInstanceCmdlets.cs]: @iantalarico This is silently continuing now. `WriteResourceExistsError(...)` writes an error record to the error stream (a suppressible non-terminating error). If we let the exception bubble up, the error would be a terminating error.

Comment (iantalarico) [reply Google.PowerShell/Compute/GceInstanceCmdlets.cs]: Do we also need to check 

... when(apiEx.HttpStatusCode == HttpStatusCode.Conflict && SilentlyContinue)?

Comment (quoctruong) [reply Google.PowerShell/Compute/GceInstanceCmdlets.cs]: @iantalarico We don't need to check for the `SilentlyContinue` parameter. It's a built-in PowerShell parameter that will suppress error record (which is written out by the cmdlet in the catch clause).

Comment (iantalarico) [reply Google.PowerShell/Compute/GceInstanceCmdlets.cs]: Ah I see, we just write the error and let PS handle it.  SGTM

File: Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs [hunk 2]
```diff
@@ -52,6 +53,72 @@ public void TestAcceptingCustomMemoryCustomCpu()
...
+            Pipeline.Commands.AddScript(
+                $"Add-GceInstance -Name instance-name -Disk ${psDiskVar}");
+            Collection<PSObject> results = Pipeline.Invoke();
+
+            // An error should be thrown (if it is a terminating error,
+            // we wouldn't even reach this point).
+            Assert.AreEqual(Pipeline.Error.Count, 1);
+
+            // Verify that the call is made.
+            instances.Verify(
```
Comment (ILMTitan) [inline Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs]: I think this verification is unnecessary. The error in the error pipeline tells us we made the correct call.

File: Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs [hunk 3]
```diff
@@ -52,6 +53,72 @@ public void TestAcceptingCustomMemoryCustomCpu()
...
+                i => i.Insert(It.IsAny<Instance>(), It.IsAny<string>(), It.IsAny<string>()),
+                apiException);
+
+            Pipeline.Commands.AddScript(
+                $"Add-GceInstance -Name instance-name -Disk ${psDiskVar}");
+            Collection<PSObject> results = Pipeline.Invoke();
+
+            // An error should be thrown (if it is a terminating error,
+            // we wouldn't even reach this point).
+            Assert.AreEqual(Pipeline.Error.Count, 1);
```
Comment (ILMTitan) [inline Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs]: Do we want to test something about the error? Perhaps test `Pipeline.Error[0].ErrorCategory == ErrorCategory.ResourceExists`?

File: Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs [hunk 4]
```diff
@@ -52,6 +53,72 @@ public void TestAcceptingCustomMemoryCustomCpu()
...
+                apiException);
+
+            Pipeline.Commands.AddScript(
+                $"Add-GceManagedInstanceGroup -Name instance-group -InstanceTemplate template -TargetSize 1");
+            Collection<PSObject> results = Pipeline.Invoke();
+
+            // An error should be thrown (if it is a terminating error,
+            // we wouldn't even reach this point).
+            Assert.AreEqual(Pipeline.Error.Count, 1);
+
```
Comment (ILMTitan) [inline Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs]: Test the error?

File: Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs [hunk 5]
```diff
@@ -52,6 +53,72 @@ public void TestAcceptingCustomMemoryCustomCpu()
...
+            // Verify that the call is made.
+            instances.Verify(
```
Comment (ILMTitan) [inline Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs]: Again, I don't think this is necessary.

**Comment by quoctruong** [comment]
Thanks! Fixed the tests.

**Pull request merged**